### ISSUE TITLE
Defining a method called "format" on an enum triggers incorrect errors

### ISSUE THREAD

**Issue by ruuda** [python/mypy#7787]
Please provide more information to help us understand the issue:

* Are you reporting a bug, or opening a feature request? I am reporting a bug.
* Please insert below the code you are checking with mypy.

    ```python
    from enum import Enum, auto

    class Widget(Enum):
        vanilla = auto()
        chocolate = auto()

        def format(self, n: int) -> str:
            return f'{self.name} {n}'

    print(Widget.vanilla.format(10))
    ```

* What is the actual behavior/output?

      repro.py:10: error: Not all arguments converted during string formatting
      Found 1 error in 1 file (checked 1 source file)

  This happens at the call to `format`.

* What is the behavior/output you expect? The file should typecheck without errors.
* What are the versions of mypy and Python you are using? Mypy 0.740 on Python 3.7.5.
  Do you see the same issue after installing mypy from Git master? Yes, on 09c1fc7a4c19fe76c2ca9f0c58eaeccfe1ef9ed5 the problem is still present.
* What are the mypy flags you are using? None, just the default.

Issue #7563 seemed superficially related, but now that I am taking a closer look, I’m not sure any more.

**Comment by ilevkivskyi** [comment]
Yes, this looks like a separate issue. I think this is relatively low priority.

**Comment by ilevkivskyi** [comment]
It looks like this works correctly now.
